service-storage/redis.ts

```diff
@@ -0,0 +1,22 @@
+import { createClient } from 'redis';
+import 'dotenv/config';
+
+export const clientRedis = createClient({
+  url: process.env.REDIS_URL,
+  password: process.env.REDIS_PASSWORD,
+});
+clientRedis.on('error', (err) => console.log('Redis Client Error', err));
+
+export const getValue = async (key: string) => {
+  const value = await clientRedis.get(key);
+  return value;
+};
+
+export const setValue = async (key: string, value: string) => {
+  await clientRedis.set(key, value, { EX: 60 * 60 * 24 }); // 24 hr expire
```

Comment:
I think we should not set expirations here it's okay to allow Redis evict keys following LRU algo, so we don't have to fetch popular contracts everyday.

Reply:
i see kub